### ISSUE TITLE
How to submit News items?

### ISSUE THREAD

**Issue by anjohnson** [epics-base/website#17]
There is no link on the News page explaining how to submit news items. The current items look to have been copied from my News page, which had about 2-3 items a year. If we want to increase the frequency of new items we probably need to make it easier for people to submit items (and we should seed the page with the kinds of things we want to see on it; Mark Rivers is about to announce a new release of Asyn for example). I found the [How to submit news to the EPICS Website](https://epics-controls.org/resources-and-support/documents/epics-website-documentation/how-to-submit-news-to-the-epics-website/) document, but that does require all EPICS news sources to sign up for an account, which I see as a discouragement.

Can I suggest putting a link on the [News and Events](https://epics-controls.org/news-and-events/) page to a form for entering a news item that the site maintainers could turn into a posting? The form page could also encourage submitters to sign up for an account, but we if we force them to do that first we probably aren't going to get as many entries.

**Comment by ralphlange** [comment]
+1
Valid request, good idea.
I will come up with a test candidate later today.

**Comment by ralphlange** [comment]
That test candidate will be a form generating a mail. To whom? Webmaster? Dedicated victim?

Independent from that technical issue is the question: What release related postings do we want to move onto the web site as opposed to tech-talk messages? Obvious options are:
1. Release notices of Base only?
2. Release notices for a defined list of "widely used modules"?
3. Release notices for all modules listed on the site?
4. Release notices for anything?

We would probably keep the cross-posting of any news items on tech talk, to get the broadest audience.
So - could the form just ask for the link to the tech-talk message to add as a news item?
And - if manual intervention is necessary anyway, couldn't we just start to add selected tech-talk content, and require using the form only for things that don't match our filters?

**Comment by ttkorhonen** [comment]
Adding a low-threshold posting form is fine. 
We even have one: "Contact us". It has the possibility to even submit files. Do we need to invent something new? We could point people to this page. Hurdle is very low. Having more pages for posting would just make it more confusing. 

While we are at this: we have discussed the policies on a couple of occasions. If changes are wanted it is fine but let us think them through well before implementing.

**Comment by ralphlange** [comment]
Having a different contact form would add a way for more required fields (e.g. link to tech-talk message, link to "read more" somewhere else) and removing fields (file upload makes no sense for a news item).

In Flamingo (database for incoming mails through contact forms) you can filter by contact form.

Re: policies. Where can I find them?

**Comment by ralphlange** [comment]
And: Sure. I will stop working on this issue until I receive clear instructions. No problem with that.

**Comment by ttkorhonen** [comment]
I hope you did not get offended! I just am concerned that first we asked people to join so that they can submit posts. Then soon after we ask them to do that by a web form. I would be rather confused. 

We have discussed the policies (or rather processes) a few times; there is no single document, only scattered notes in meeting minutes and emails. The most recent discussion was about the contributor vs author roles, and the review workflow. I can try to put together a description of what I think we have agreed on so far. We can take a look together then to see if we agree.

**Comment by ralphlange** [comment]
I completely agree with your argumentation, and was never planning to publicly change this without discussion. Note that my line of arguments in my 2nd comment goes in the direction of not using a form based approach.
All I started was to provide a non-public sample implementation using the existing plugins to have something tangible to talk about.

I can't remember we talked about our policies for news items on the site - that should be clarified before doing anything publicly visible, and that's the discussion I was hoping to start with my comment above.

Btw. that existing contact form is the one I fixed and activated last week. It seems to be aligned better with whatever policies that might apply... But I can easily roll it back to the broken state if you prefer. :-P

**Comment by ttkorhonen** [comment]
Sorry, I forgot the point about the contact form. Thank you for fixing it, please do not even think of rolling back.

What I refer to here as news items are those short posts that appear on the "News and Events" page. Those can be edited by "contributor"s as the first right they have after registration and approval. All posts by contributors have to go to the "simple approval" workflow. Approval can be done by Administrators or Editors currently. After approval, posts go the website. Approver can also reject the post or ask the contributor to revise the text. Or edit it him/herself.
Now the issue to be discussed is if there should be a parallel way to do this, and how should it look like. Also the other things like linking to a tech-talk post, etc. are new and have not been defined yet. 
It looks like a good idea to ask for a link to the tech-talk article. However, then somebody still has to edit the post. On the other hand we can just pick items from tech-talk and add them as posts. This is how the post for the Australian EPICS meeting was added. Today there are already three announcements on tech-talk: PCAS, ASYN and quadEM. These would be good to have on the website. There is only one drawback: the fall meeting announcement will go down the list. Maybe we need to change the "Past Meetings" item to "Collaboration Meetings" and add there a section for "Upcoming Meetings" instead of past only.
This got a bit long...my apologies.

**Comment by ralphlange** [comment]
As we have a lot of modules, I would suggest limiting the release posts to option 1 or 2 of my above comment.
Otherwise **everything** else on the news and events will go down the drain.

**Comment by ttkorhonen** [comment]
This is reasonable. Of the three above, Asyn release would qualify, PCAS maybe?, quadEM probably not.

**Comment by anjohnson** [comment]
When talking about items getting pushed down the list by newer items, are you just concerned about the front page (where we only show the top 3 items) or the whole News and Events page? Would it be possible to add a flag that marks specific news items as "show me on the front page"? Some bulletin-board systems provide the ability to pin an item to the top of the list of discussions, which is similar to what I'm suggesting, or maybe this might be done with a tag?

Just an idea, don't know if it's possible...

**Comment by ralphlange** [comment]
So my suggestion would be:
* Put "News and Events" in a category separate from "Release Announcements".
* Add "frontpage" as a tag to the items that we want to see there. 
* Add a filter to the frontpage teaser to only show items tagged "frontpage".
* Make the "News" page display two columns: "News and events" left and "Releases" right.
* Create a template post that can be used for release announcements & document this.

Comments anyone?

**Comment by anjohnson** [comment]
@ralphlange That seems like it should cover almost everything. If we ever create a more comprehensive software database to replace the supported hardware/software lists the releases column might get replaced (or fed) by entries into that database (but I guess that's still a bit of a pipe-dream of mine).

**Comment by anjohnson** [comment]
On the [News and Events](https://epics-controls.org/news-and-events/) page I just noticed that the layout for the latest (Asyn R3-14) item was different than the others; there was no margin between the graphic and the text. Looking at the post in more detail the older items were structured differently; the image should not be pasted inside the text box where it was put for the Asyn item, it needs to be set as the "Featured Image" for the post, which is selected using a box at the bottom of the right-hand column of the Edit Post page (this is actually documented on the "How to submit" page). I just changed the Asyn post to match (I also found a media item called "modules" which I used instead, I assume that's what this logo was meant for).

**Comment by ralphlange** [comment]
That's why I added the fifth item in my list: we need a template post to make things easier.
If @ttkorhonen was to support the aforementioned list, I would consider taking the ticket. :-)

**Comment by ralphlange** [comment]
Just to be clear:
@ttkorhonen - if you support this idea (the five item list above), I will take the ticket and implement.

**Comment by ttkorhonen** [comment]
Yes, I do. Looks good, please go ahead and implement. Sorry for the delay, I was occupied with other stuff.

**Comment by ralphlange** [comment]
- [x] Put "News and Events" in a category separate from "Release Announcements".
- [x] Add "frontpage" as a tag to the items that we want to see there.
- [x] Add a filter to the frontpage teaser to only show items tagged "frontpage".
- [x] Make the "News" page display two columns: "News and events" left and "Releases" right.
- [ ] Create a template post that can be used for release announcements
- [ ] Add how to use the template to website documentation.

**Comment by anjohnson** [comment]
@ralphlange Your new News page looks good on a regular browser, but the two Post columns need side margins adding — when I tell Chrome to display this for a "device" the text hits the RHS of the display and on the LHS the green boxes and vertical lines are getting clipped.

**Comment by ralphlange** [comment]
Hm. Same is true for the original current 1-column news page.
Does that make my new version good enough?

**Comment by ralphlange** [comment]
I think it justifies creating another ticket.

**Comment by anjohnson** [comment]
Agreed to both comments. The same solution would probably apply to either version then, so yes your version looks fine by me.
